### PR TITLE
Add redhat internal tasks that setup repo

### PR DESCRIPTION
**Pull request by vasukulkarni**
setup repo based on config items, either use
cdn or compose sytle urls or plain repo's.

Still testing with other change,

Signed-off-by: Vasu Kulkarni <vasu@redhat.com>

### CODE REVIEW

**Comment by zmc** [comment]
Curious - why not take advantage of the `downstream_setup` role that Andrew created ages ago? Wasn't it created for exactly this purpose?

**Comment by GregMeno** [comment]
@vasukulkarni when do you think this will be out of [DNM] ?

**Comment by vasukulkarni** [comment]
I have this already working, need to cleanup commits, setting up meeting with zack, ken to get most of issues sorted out and merged.

File: teuthology/task/internal/redhat.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+@contextlib.contextmanager
+def setup_cdn_repo(ctx, config):
+    """
+     setup repo if set_cdn_repo exists in config
+     redhat:
+      set_cdn_repo:
+        rhbuild: 2.0 or 1.3.2 or 1.3.3
+    """
+    # do import of tasks here since the qa task path should be set here
+    if ctx.config.get('redhat').get('set-cdn-repo', None):
```
Comment (zmc) [inline teuthology/task/internal/redhat.py]: If `redhat` is undefined, you'll still get an `AttributeError` here, so the first `get()` won't help much

**Pull request merged**